inputters/base.lua

```diff
@@ -33,24 +33,11 @@ end
 
 function inputter:requireClass (tree)
   local root = SILE.documentState.documentClass == nil
-  local document
-  if root then
-    for _, leaf in ipairs(tree) do
-      if leaf.lno then
-        if document then
-          SU.error("Document has more than one parent node!")
-        end
-        if leaf.command == "sile" or leaf.command == "document" then
-          document = leaf
-        end
-      end
-    end
-    if not document then
-      SU.error("This isn't a SILE document!")
-    end
-    self:classInit(document.options or {})
-    self:preamble()
+  if root and tree.command ~= "sile" and tree.command ~= "document" then
+    SU.error("This isn't a SILE document!")
   end
+  self:classInit(tree.options or {})
```

Comment:
I think this is wrong and SILE <= 0.14.4 was actually right here:
It make sense that we only need to init a class and check preambles if the document is "root", i.e. not inputted while there's already a class, i.e. typically from CLI.


 local root = SILE.documentState.documentClass == nil
  if root then
    if #tree ~= 1
      or (tree[1].command ~= "sile" and tree[1].command ~= "document") then
      SU.error("This isn't a SILE document!")
    end
    self:classInit(tree[1].options or {})
    self:preamble()
  end

Reply:
Good catch, the nesting of `if root` is borked here, I combined two conditionals incorrectly.